kqueue.go

```diff
@@ -231,7 +239,16 @@ func (w *Watcher) addWatch(name string, flags uint32) (string, error) {
 
 	if !alreadyWatching {
 		w.mu.Lock()
+		parentName := filepath.Dir(name)
 		w.watches[name] = watchfd
+
+		watchesByDir, ok := w.watchesByDir[parentName]
```

Comment:
we don't need to do anything of this stuff if `name` ends in the file separator as that means it is the root directory and the root directory as no parent. Likewise a similar thing should happen in Remove().

Reply:
You know what, I just realized thats how it works. If you type `cd ..` at the root of the FS, you remain in the root so in a certain sense, the root is a parent of itself. 🤔

Reply:
So lets just leave it the way it is.